db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -1019,7 +1019,7 @@ CostEstimate estimateCostHelper(OptimizablePredicateList predList,
                 getLanguageConnectionContext().getOptimizerFactory().getIndexBatchSize(),
                 getLanguageConnectionContext().getOptimizerFactory().getIndexLookupBlocks(),
                 forUpdate(),
-                dataSetProcessorType.isOlap(),
+                optimizer.isForSpark(),
```

Comment:
I'm not sure I understand why this was changed.

Reply:
Previously, if you set `useOlap=true` on a specific `FROM` table, then not all `FromBaseTable` will be affected, i.e. some of them will still have `useOlap=false` which doesn't make sense, this fixes that.

Reply:
`dataSetProcessorType` in `FromBaseTable` is only set by hint on this specific table. Like @hatyo mentioned, this hint is propagated to `optimizer.isForSpark()`, but not all other `dataSetProcessorType`. If we do it in the old way, some tables may get cost estimate using OLTP formula while others get costs using OLAP formula.